client/src/app/editor/file-outputs/file-outputs.component.ts

```diff
@@ -57,30 +60,64 @@ export class FileOutputsComponent implements OnChanges, OnInit {
   ngOnInit() {
     const outputFileId = this.route.snapshot.queryParamMap.get('preview');
 
-    if (outputFileId) {
-      this.hasOutput
-        .pipe(
-          filter(hasOutput => hasOutput),
-          mergeMap(() => this.dataSource.connect()),
-          mergeMap(outputFiles => outputFiles),
-          filter(outputFile => outputFile.id === outputFileId),
-          filter(outputFile => isImage(outputFile.name)),
-          take(1)
+    const executionSelected$ = this.executionId$.pipe(
+      switchMap(executionId => this.labExecutionService.executionExists(executionId).pipe(catchError(() => of(false)))),
+      filter(Boolean)
+    );
+
+    const executionsAdded$ = this.labExecutionService.observeExecutionsForLab(this.lab).pipe(
+      mergeMap(executions => executions),
+      filter(labExecution => labExecution.id === this.executionId),
+      distinctUntilChanged((a, b) => a.id === b.id),
+      skip(1)
+    );
+
+    merge(executionSelected$, executionsAdded$)
+      .pipe(switchMap(() => this.labExecutionService.observeExecution(this.executionId)))
+      .subscribe(execution => {
+        this.isExecuting = execution.status === ExecutionStatus.Executing;
+      });
+
+    this.isLoading$ = merge(
+      this.executionId$.pipe(mapTo(true)),
+      this.executionId$.pipe(
+        switchMap(executionId => this.outputFilesService.hasOutputFiles(executionId)),
+        delay(4000)
```

Comment:
Still have to try it locally but isn't 4 seconds a bit much?

Reply:
This is essentially a timeout. This is not artificially slowing down the loading process of the output files. The `Loading output files...` inside the `file-outputs-table` component is only shown if it's loading **and** if there are no output files. If we already have output files before the 4 seconds, we hide that message and show the output files in the table.

Reply:
A comment about that might be useful